CoThings/Views/Settings/QRCodeView.swift

```diff
@@ -74,25 +80,64 @@ class QRCodeViewController: UIViewController {
         previewLayer.frame = view.layer.bounds
         previewLayer.videoGravity = .resizeAspectFill
         view.layer.addSublayer(previewLayer)
-
-        captureSession.startRunning()
+        
+        let titleLabel = UILabel()
+        titleLabel.text = "Scan QR code"
+        titleLabel.font = .systemFont(ofSize: 20, weight: .heavy)
+        titleLabel.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(titleLabel)
+        titleLabel.topAnchor.constraint(equalTo: view.topAnchor, constant: 20).isActive = true
+        titleLabel.centerXAnchor.constraint(equalTo: view.centerXAnchor).isActive = true
+        
+        let dismissButton = UIButton()
+        let crossImage = UIImage(systemName: "multiply", withConfiguration: UIImage.SymbolConfiguration(pointSize: 35, weight: .regular, scale: .large))
+        dismissButton.setImage(crossImage, for: .normal)
+        dismissButton.tintColor = .white
+        dismissButton.addTarget(self, action: #selector(onTapDismissButton), for: .touchUpInside)
+        dismissButton.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(dismissButton)
+        dismissButton.topAnchor.constraint(equalTo: view.topAnchor, constant: 15).isActive = true
+        dismissButton.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 15).isActive = true
+        dismissButton.heightAnchor.constraint(equalToConstant: 35).isActive = true
+        dismissButton.widthAnchor.constraint(equalTo: dismissButton.heightAnchor).isActive = true
+        
+        
+        qrCodeNotValidLabel.text = ""
+        qrCodeNotValidLabel.font = .systemFont(ofSize: 15, weight: .bold)
+        qrCodeNotValidLabel.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(qrCodeNotValidLabel)
+        qrCodeNotValidLabel.centerXAnchor.constraint(equalTo: view.centerXAnchor).isActive = true
+        qrCodeNotValidLabel.centerYAnchor.constraint(equalTo: view.centerYAnchor).isActive = true
+        
+
+        sessionQueue.async {
+            self.captureSession.startRunning()
+        }
     }
 
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
 
         if (captureSession?.isRunning == false) {
```

Comment:
Checking the equality of a boolean feels weird, instead of it `!captureSession?.isRunning` feels more natural.